aioxmpp/bookmarks/xso.py

```diff
@@ -30,7 +31,96 @@
 namespaces.xep0048 = "storage:bookmarks"
 
 
-class Conference(xso.XSO):
+class Bookmark(xso.XSO):
+    """A bookmark XSO abstract base class.
+
+    Every XSO class registered as child of :class:`Storage` must be
+    a :class:`Bookmark` subclass.
+
+    Bookmarks must provide the following interface:
+
+    .. autoproperty:: primary
+
+    .. autoproperty:: secondary
+
+    .. autoproperty:: name
+
+    Equality is defined in terms of those properties:
+
+    .. automethod:: __eq__
+
+    It is highly recommenden not to redefine :meth:`__eq__` in a
+    subclass, if you do so make sure that the following law
+    holds for :meth:`__eq__`, :attr:`primary` and :attr:`secondary`:
+
+        (type(a) == type(b) and
+         a.primary == b.primary and
+         a.secondary == b.secondary)
```

Comment:
Is the formatting here intended? It doesn’t look that nice when rendered in HTML, so maybe use ``::`` to make it code formatted:

>     holds for :meth:`__eq__`, :attr:`primary` and :attr:`secondary`::
>         
>         (type(a)…
>
>     if and only if::